src/i2c.rs

```diff
@@ -42,8 +42,12 @@ where
     }
 
     /// Initializes the BME280
-    pub fn init<D: DelayUs>(&mut self, delay: &mut D) -> Result<(), Error<I2C::Error>> {
-        self.common.init(delay)
+    pub fn init<D: DelayUs>(
+        &mut self,
+        delay: &mut D,
+        config: Configuration,
+    ) -> Result<(), Error<I2C::Error>> {
+        self.common.init(delay, config)
```

Comment:
Do you mind leaving the current 'init' function as is that will default to the current configuration and add new function. Perhaps 'init_with_config' that will take in the configuration struct?